impl/src/test/java/slack/SlackClientImplTest.java

```diff
@@ -0,0 +1,178 @@
+import com.github.seratch.jslack.api.methods.request.channels.ChannelsRepliesRequest;
+import com.github.seratch.jslack.api.methods.request.chat.ChatPostMessageRequest;
+import com.github.seratch.jslack.api.methods.request.users.UsersInfoRequest;
+import com.github.seratch.jslack.api.methods.request.users.UsersLookupByEmailRequest;
+import com.github.seratch.jslack.api.methods.response.channels.ChannelsRepliesResponse;
+import com.github.seratch.jslack.api.methods.response.channels.UsersLookupByEmailResponse;
+import com.github.seratch.jslack.api.methods.response.chat.ChatPostMessageResponse;
+import com.github.seratch.jslack.api.methods.response.users.UsersInfoResponse;
+import org.apache.log4j.Appender;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import se.fortnox.reactivewizard.test.LoggingMockUtil;
+
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.mockito.Mockito.*;
+import static se.fortnox.reactivewizard.test.TestUtil.matches;
+
+public class SlackClientImplTest {
+
+    private SlackConfig slackConfig;
+    private SlackClient slackClient;
+    private Slack slack = mock(Slack.class);
+    private MethodsClient methodsClient;
+    private Appender appender;
+
+    @Before
+    public void beforeEach() throws NoSuchFieldException, IllegalAccessException {
+        slackConfig = new SlackConfig();
+        slackConfig.setApiToken("1234");
+        methodsClient = mock(MethodsClient.class);
+        when(slack.methods()).thenReturn(methodsClient);
+        slackClient = new SlackClientImpl(slack);
+        appender = LoggingMockUtil.createMockedLogAppender(SlackClientImpl.class);
+    }
+
+    @After
+    public void afterEach() throws NoSuchFieldException, IllegalAccessException {
+        LoggingMockUtil.destroyMockedAppender(appender, OpenIdValidator.class);
+    }
+
+    @Test
+    public void shouldCallUsersLookupByEmail() throws Exception {
+        // given a user lookup request
+        UsersLookupByEmailRequest usersLookupByEmailRequest = UsersLookupByEmailRequest.builder()
+            .email("email@to.lookup")
+            .token(slackConfig.getApiToken())
+            .build();
+
+        // when we look up user
+        UsersLookupByEmailResponse usersLookupByEmailResponse = new  UsersLookupByEmailResponse();
+        when(methodsClient.usersLookupByEmail(usersLookupByEmailRequest)).thenReturn(usersLookupByEmailResponse);
+
+        UsersLookupByEmailResponse response = slackClient.usersLookupByEmail(usersLookupByEmailRequest).toBlocking().single();
+        // then the real method should be invoked
+        verify(methodsClient, times(1)).usersLookupByEmail(usersLookupByEmailRequest);
+        // and the token in the config should be used
+        assertThat(usersLookupByEmailRequest.getToken()).isEqualTo(slackConfig.getApiToken());
```

Comment:
Isn't this just testing the builder? We could use an argumentcaptor to get at the request that is constructed.
The same goes for the test cases below.

Reply:
Yes, you are right. Will remove the line.